src/elf/symver.rs

```diff
@@ -0,0 +1,902 @@
+        fn next(&mut self) -> Option<Self::Item> {
+            if self.index >= self.count {
+                None
+            } else {
+                self.index += 1;
+
+                // Safe to unwrap as the length of the byte slice was validated in VerdefSection::parse.
+                let ElfVerdef {
+                    vd_version,
+                    vd_flags,
+                    vd_ndx,
+                    vd_cnt,
+                    vd_hash,
+                    vd_aux,
+                    vd_next,
+                } = self.bytes.pread_with(self.offset, self.ctx.le).unwrap();
+
+                // Get a slice of bytes of the `Verdaux` entries.
+                //
+                // | Verdef | .. | Verdef | Verdaux | Verdaux | .. | Verdef | ..
+                // ^-------------^
+                //  offset       ^---------^
+                //                vd_aux
+                //               ^---------------------------------^
+                //                 vd_next
+                //
+                // Safe to unwrap as the length of the byte slice was validated in VerdefSection::parse.
+                let len = if vd_next > 0 {
+                    (vd_next - vd_aux) as usize
+                } else {
+                    // For the last entry, ElfVerdef->vd_next == 0.
+                    // Therefore we compute the remaining length of the bytes buffer.
+                    self.bytes.len() - self.offset - vd_aux as usize
+                };
+                let bytes: &'a [u8] = self
+                    .bytes
+                    .pread_with(self.offset + vd_aux as usize, len)
+                    .unwrap();
+
+                // Bump the offset to the next ElfVerdef entry.
+                self.offset += vd_next as usize;
+
+                Some(Verdef {
+                    vd_version: vd_version as usize,
+                    vd_flags: vd_flags as usize,
+                    vd_ndx: vd_ndx as usize,
+                    vd_cnt: vd_cnt as usize,
+                    vd_hash: vd_hash as usize,
+                    vd_aux: vd_aux as usize,
+                    vd_next: vd_next as usize,
+                    bytes,
+                    ctx: self.ctx,
+                })
+            }
+        }
+
+        #[inline]
+        fn size_hint(&self) -> (usize, Option<usize>) {
+            let len = self.count - self.index;
+            (len, Some(len))
+        }
+    }
+
+    impl ExactSizeIterator for VerdefIter<'_> {}
+
+    impl FusedIterator for VerdefIter<'_> {}
+
+    /// An ELF [Version Definition][lsb-verdef] entry .
+    ///
+    /// [lsb-verdef]: https://refspecs.linuxbase.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/symversion.html#VERDEFENTRIES
+    #[derive(Debug)]
+    pub struct Verdef<'a> {
+        pub vd_version: usize,
+        pub vd_flags: usize,
+        pub vd_ndx: usize,
+        pub vd_cnt: usize,
+        pub vd_hash: usize,
+        pub vd_aux: usize,
+        pub vd_next: usize,
```

Comment:
I don't think `usize` is appropriate for all of these, since they don't vary between 32-bit and 64-bit ELF. e.g. `vd_hash` is only ever going to be used as a `u32`. I suggest only using `usize` for fields that the user will commonly use as an index (I'm not sure which fields qualify for that, but I doubt it is many).

Reply:
Fair point, and that counts for all the other structures exposed.